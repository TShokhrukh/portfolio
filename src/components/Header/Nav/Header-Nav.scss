.Header-Nav {
  display: flex;
  align-items: center;
  margin-left: auto;
}
