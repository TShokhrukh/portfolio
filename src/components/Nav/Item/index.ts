export { NavItem as default } from './Nav-Item'
export * from './Nav-Item'
