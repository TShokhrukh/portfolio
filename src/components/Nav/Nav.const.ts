import { cn } from '@bem-react/classname'

export const cnNav = cn('Nav')
