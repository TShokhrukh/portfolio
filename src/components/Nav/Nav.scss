
.Nav-List {
  margin: 0;
  list-style: none;
}

.Nav-ItemLink {
  font-size: 18px;
  color: rgba($white, 0.5);
}

.Nav-Item_active .Nav-ItemLink {
  color: rgba($purple, 0.5);
} 