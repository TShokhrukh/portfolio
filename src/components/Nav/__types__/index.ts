export * from './Nav.types'
export * from './Nav_direction_horizontal.types'
export * from './Nav_direction_vertical.types'
