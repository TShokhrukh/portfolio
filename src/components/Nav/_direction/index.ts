export * from './Nav_direction_horizontal'
