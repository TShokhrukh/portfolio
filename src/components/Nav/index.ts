export * from './build'
export * from './__types__'
export { Nav as NavBase } from './Nav'
