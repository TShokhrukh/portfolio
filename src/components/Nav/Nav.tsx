import React from 'react'
import { cnNav } from './Nav.const'
import NavItem from './Item'
import type { INavProps } from './__types__'
import { NavContext } from './libs/context'

import './Nav.scss'

export const Nav: React.FC<INavProps> = ({
  className,
  links,
  active,
  ...props
}) => {
  return (
    <NavContext.Provider value={{ active }}>
      <nav {...props} className={cnNav({}, [className])}>
        <ul className={cnNav('List')}>
          { links.map((l) => (<NavItem key={l.href} link={l} />)) }
        </ul>
      </nav>
    </NavContext.Provider>
  )
}
