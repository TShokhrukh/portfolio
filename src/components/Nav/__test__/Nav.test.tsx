/* global test, expect, describe */
import React from 'react'
import { render, screen } from '@testing-library/react'
import { NavWrapper } from './NavWrapper'
import { Nav } from '../Nav'

import type { ILink } from '../__types__'

describe('Nav', () => {
  test('must return the component template', () => {
    const links: ILink[] = [
      {
        href: 'one',
        label: 'link-1'
      },
      {
        href: 'two',
        label: 'link-2'
      }
    ]
    render(
      <NavWrapper>
        <Nav links={links} active='two' data-testid='nav' />
      </NavWrapper>
    )

    expect(screen.getByTestId('nav')).toMatchSnapshot()
  })
})
