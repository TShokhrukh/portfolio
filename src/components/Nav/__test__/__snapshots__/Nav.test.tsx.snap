// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav must return the component template 1`] = `
<nav
  class="Nav"
  data-testid="nav"
>
  <ul
    class="Nav-List"
  >
    <li
      class="Nav-Item"
    >
      <a
        class="Link Nav-ItemLink"
        href="one"
      >
        link-1
      </a>
    </li>
    <li
      class="Nav-Item Nav-Item_active"
    >
      <a
        class="Link Nav-ItemLink"
        href="two"
      >
        link-2
      </a>
    </li>
  </ul>
</nav>
`;
